src/Vault.sol

```diff
@@ -1173,6 +1194,9 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, Ownable {
     if (_withdrawableAssetsAfter < _expectedWithdrawableAssets)
       revert YVWithdrawableAssetsLTExpected(_withdrawableAssetsAfter, _expectedWithdrawableAssets);
 
+    if (_assets > maxMint(_receiver))
+      revert MintMoreThanMax(_receiver, _assets, maxMint(_receiver));
```

Comment:
Why do we need this?

- `_yieldVault.deposit()` above should prevent depositing more than the max
- `_mint()` below this should prevent minting more shares than max